packages/date-picker/src/panel/date.vue

```diff
@@ -479,11 +524,25 @@
       },
 
       isValidValue(value) {
-        return value && !isNaN(value) && (
-          typeof this.disabledDate === 'function'
-            ? !this.disabledDate(value)
-            : true
-        ) && this.checkDateWithinRange(value);
+        return this.isValidDate(value) && this.checkTimeWithinRange(value);
+      },
+
+      isValidDate(date) {
```

Comment:
Consolidate logic in `isValidDate` and `isValidValue` methods (the latter also checks the time for date & time fields)